### PR TITLE
Reserve proofs

### PR DESCRIPTION
**Pull request by cryptochangements34**
Add a new tab under "advanced" called "Reserve proofs" where the user can create and verify reserve proofs which prove that a user owns at least X amount of Monero

### CODE REVIEW

File: LeftPanel.qml [hunk 1]
```diff
@@ -472,6 +474,29 @@ Rectangle {
                 color: "#313131"
                 height: 1
             }
+            // ----------- Reserve Proofs tab -------------
+            MenuButton {
+                id: reserveButton
+                anchors.left: parent.left
+                anchors.right: parent.right
+                text: qsTr("Reserve Proofs") + translationManager.emptyString
+                symbol: qsTr("K") + translationManager.emptyString
```
[the remark this answers is not in the document]
Comment (cryptochangements34) [reply LeftPanel.qml]: Yes, thanks

**Comment by dEBRUYNE-1** [comment]
@cryptochangements34 - Have you already looked at stoffu's comments?

**Comment by dEBRUYNE-1** [comment]
Ping @cryptochangements34.

File: LeftPanel.qml (hunk 1 above)
[the remark this answers is not in the document]
Comment (xmrdsc) [reply LeftPanel.qml]: Fixed in #1927

File: LeftPanel.qml [hunk 2]
```diff
@@ -68,6 +69,7 @@ Rectangle {
         else if(pos === "AddressBook") menuColumn.previousButton = addressBookButton
         else if(pos === "Mining") menuColumn.previousButton = miningButton
         else if(pos === "TxKey")  menuColumn.previousButton = txkeyButton
+        else if(pos === "ReserveProofs")  menuColumn.previousButton = reserveButton
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply LeftPanel.qml]: Fixed in #1927

File: main.qml [hunk 3]
```diff
@@ -102,6 +102,7 @@ ApplicationWindow {
         if(seq === "Ctrl+S") middlePanel.state = "Transfer"
         else if(seq === "Ctrl+R") middlePanel.state = "Receive"
         else if(seq === "Ctrl+K") middlePanel.state = "TxKey"
+        else if(seq === "Ctrl+W") middlePanel.state = "ReserveProof"
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply main.qml]: Fixed in #1927

File: pages/ReserveProof.qml [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+            anchors.top: parent.top
+            spacing: 20 * scaleRatio
+
+            Label {
+                fontSize: 24 * scaleRatio
+                text: qsTr("Get Reserve Proof") + translationManager.emptyString
+            }
+
+            Text {
+                text: qsTr("Generate a verifiable cryptographic proof that your own at least a certain amount of Monero") + translationManager.emptyString
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply pages/ReserveProof.qml]: Fixed in #1927

File: pages/ReserveProof.qml [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+            }
+
+            StandardButton {
+                anchors.left: parent.left
+                anchors.topMargin: 17 * scaleRatio
+                width: 60 * scaleRatio
+                text: qsTr("Generate") + translationManager.emptyString
+                enabled: amountLine.text > 0
+                onClicked: {
+                  informationPopup.title  = qsTr("Reserve proof") + translationManager.emptyString;
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply pages/ReserveProof.qml]: Fixed in #1927

File: pages/ReserveProof.qml [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+            }
+
+            StandardButton {
+                anchors.left: parent.left
+                anchors.topMargin: 17 * scaleRatio
+                width: 60 * scaleRatio
+                text: qsTr("Verify") + translationManager.emptyString
+                enabled: TxUtils.checkAddress(addressLine.text, appWindow.persistentSettings.nettype)
+                onClicked: {
+                  informationPopup.title  = qsTr("Proof Result") + translationManager.emptyString;
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply pages/ReserveProof.qml]: Fixed in #1927

File: pages/ReserveProof.qml [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+                id: amountColumn
+                Label {
+                    id: amountLabel
+                    text: qsTr("Amount") + translationManager.emptyString
+                    width: mainLayout.labelWidth
+                }
+
+                LineEdit {
+                    id: amountLine
+                    placeholderText: qsTr("Amount to prove") + translationManager.emptyString
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply pages/ReserveProof.qml]: Fixed in #1927

File: pages/ReserveProof.qml [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+                width: 60 * scaleRatio
+                text: qsTr("Verify") + translationManager.emptyString
+                enabled: TxUtils.checkAddress(addressLine.text, appWindow.persistentSettings.nettype)
+                onClicked: {
+                  informationPopup.title  = qsTr("Proof Result") + translationManager.emptyString;
+                  var result;
+                  if(appWindow.currentWallet.checkReserveProof(addressLine.text, proveMessageLine.text, signatureLine.text))
+                      result = "True Proof";
+                  else
+                      result = "False Proof";
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply pages/ReserveProof.qml]: Fixed in #1927

**Comment by xmrdsc** [comment]
Lets continue in #1927

+invalid

**Pull request closed without merging**